### ISSUE TITLE
Parameters for individual decks don't work

### ISSUE THREAD

**Issue by Expertium** [open-spaced-repetition/fsrs4anki#41]
**Describe the bug**
You can specify parameters for each deck, but right now only global parameters affect scheduling.

**To Reproduce**
1. Go to deck options and post code from fsrs4anki_scheduler.js in the window at the bottom of the settings
2. Add this chunk:
if (document.getElementById('deck') !== null) {

    const deck_name = document.getElementById('deck').innerHTML;
    // parameters for a specific deck
    if (deck_name == "DeckName") {
        var f_s = [0.7712,0.837];
        var f_d = [1.0018,-0.9006,-1.1375,0.0935];
        var s_w = [3.0531,-0.8949,-0.1383,1.3434,2.1846,-0.2868,0.4257,1.245];

        requestRetention = 0.80;
        maximumInterval = 3650;
        easyBonus = 1.5;
        hardInterval = 1.2;
    }
}
3. Try doing reviews with the above piece of code and without it (with global parameters)
4. Compare the intervals, they should be different since the parameters are different

**Expected behavior**
Different sets of parameters should produce different intervals.

**Environment**
 - Anki version: 2.1.55
 - OS: Windows 10
 - Scheduler version: v2.0.3
 - Optimizer version: v2.0.3

**Additional context**
Here's all of my code: 
[code.txt](https://github.com/open-spaced-repetition/fsrs4anki/files/9667781/code.txt)

**Comment by L-M-Sherlock** [comment]
Can you open the inspector and run the code to this line? It will show the deck name that the code actually fetchs.

![image](https://user-images.githubusercontent.com/32575846/192918540-5d1de839-56a7-4b37-8910-5a3dade99b5b.png)

**Comment by Expertium** [comment]
Yep, it uses the correct deck name, but not the correct parameters.
![1](https://user-images.githubusercontent.com/83031600/193000168-bdba5283-c5c2-423e-bb73-8819d28c8c52.png)

**Comment by L-M-Sherlock** [comment]
Was `var f_s = [0.7249,0.8026]` skipped?

**Comment by Expertium** [comment]
My global parameters are 
```
var f_s = [0.4112,1.5968];
var f_d = [1.0199,-0.4685,-1.7075,0.0714];
var s_w = [3.5014,-1.3554,-0.138,1.7324,2.9985,-0.0531,0.9896,1.3442];
```
My parameters for that deck are:
```
var f_s = [0.7249,0.8026];
var f_d = [1.0007,-0.6561,-1.2081,0.0112];
var s_w = [3.2511,-0.8703,-0.0277,1.5211,2.1651,-0.2846,0.5221,1.2443]
```
"Scope" shows the wrong parameters, but the right deck name.
![1](https://user-images.githubusercontent.com/83031600/193011149-7f852067-59e4-4374-b1c1-b785e457c8af.png)

I'm using scheduler v2.0.3

**Comment by L-M-Sherlock** [comment]
Can you step over the line 31-38, then to check the scope?

**Comment by Expertium** [comment]
![image](https://user-images.githubusercontent.com/83031600/193028103-a63b5d25-f187-446d-816b-087e3b031c53.png)

**Comment by Expertium** [comment]
Huh, weird. I'll test it a bit more later, now that I've disabled button king add-on.

**Comment by L-M-Sherlock** [comment]
Can you follow the action in this GIF?

![debug](https://user-images.githubusercontent.com/32575846/193041981-91a58eb7-b4cf-4d47-bd8c-80b38574924e.gif)

I need `easy_interval`, `good_interval` and `hard_interval` in your case.

**Comment by Expertium** [comment]
Here
![image](https://user-images.githubusercontent.com/83031600/193064985-0c151194-f0e5-4ba4-a099-db145acca230.png)
I once again tried enabling and disabling "the king of button add-ons", it doesn't make a difference.

**Comment by L-M-Sherlock** [comment]
I add a new example of setting parameters for a specific deck:

<img width="772" alt="image" src="https://user-images.githubusercontent.com/32575846/193075102-4ededc7b-d248-4ddb-9013-b475410ffe7b.png">

**Comment by Expertium** [comment]
![image](https://user-images.githubusercontent.com/83031600/193076418-ab5d2e3c-782e-4ee4-9bb0-0a2585df45ba.png)
![image](https://user-images.githubusercontent.com/83031600/193076612-e9686d6f-f617-45eb-88b3-66ff5922dc65.png)
Nope, intervals still don't change, no matter what parameters I use and no matter whether I even use code from fsrs4anki_scheduler.js.

**Comment by L-M-Sherlock** [comment]
Do you click `show answer` before the code run to the end?

**Comment by L-M-Sherlock** [comment]
You can click the blue arrow to run to the end of the code. Then you click the `show answer`, and the interval may be different.
<img width="400" alt="image" src="https://user-images.githubusercontent.com/32575846/193077656-b52aa3f7-5dfe-4251-94c3-8ed5a0d0d101.png">

**Comment by Expertium** [comment]
Nope, still the same.

**Comment by L-M-Sherlock** [comment]
So weird😭.

**Comment by Expertium** [comment]
Do you not have the same issue?
I have one last idea in mind - I'll try disablign every single add-on, and see if that helps.

**Comment by Expertium** [comment]
![image](https://user-images.githubusercontent.com/83031600/193080275-89df05de-e83c-4352-987b-81740f786122.png)
Ok, I'm getting somewhere. I will turn add-ons on one-by-one, and see if maybe one of them breaks the whole thing (seems liek that's what's happening)

**Comment by L-M-Sherlock** [comment]
I'm looking forward to your good news!

**Comment by L-M-Sherlock** [comment]
I add the incompatible add-ons to the list at [Compatibility](https://github.com/open-spaced-repetition/fsrs4anki/wiki/Compatibility).
